myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+            "food[description]": description,
+            "weight[serving_size]": serving_size,
+            "servingspercontainer": f"{servingspercontainer}",
+            "nutritional_content[calories]": f"{calories}",
+            "nutritional_content[sodium]": f"{sodium}",
+            "nutritional_content[fat]": f"{fat}",
+            "nutritional_content[potassium]": f"{potassium}",
+            "nutritional_content[saturated_fat]": f"{saturated_fat}",
+            "nutritional_content[carbs]": f"{carbs}",
+            "nutritional_content[polyunsaturated_fat]": f"{polyunsaturated_fat}",
+            "nutritional_content[fiber]": f"{fiber}",
+            "nutritional_content[monounsaturated_fat]": f"{monounsaturated_fat}",
+            "nutritional_content[sugar]": f"{sugar}",
+            "nutritional_content[trans_fat]": f"{trans_fat}",
+            "nutritional_content[protein]": f"{protein}",
+            "nutritional_content[cholesterol]": f"{cholesterol}",
+            "nutritional_content[vitamin_a]": f"{vitamin_a}",
+            "nutritional_content[calcium]": f"{calcium}",
+            "nutritional_content[vitamin_c]": f"{vitamin_c}",
+            "nutritional_content[iron]": f"{iron}",
+            "food_entry[quantity]": "1.0",
+            "food_entry[meal_id]": "0",
+            "addtodiary": "no",
+            "preserve_exact_description_and_brand": "true",
+            "continue": "Save",
+        }
+        # Make entry public if requested, Hint: submit "sharefood": 0 also generates a public db entry, so only add
+        # "sharefood"" if really requested
+        if sharepublic:
+            data["sharefood"] = 1
+
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_POST_PATH)
+        result = self.session.post(
+            url,
+            data,
+        )
+        if result.status_code == 200:
+            document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+            try:
+                if not document.xpath(
+                    # If list is empty there should be no error, could be replaced with assert
+                    "//*[@id='errorExplanation']/ul/li"
+                ):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
+                else:  # Error occurred
+                    error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+                    error = error.replace("Description ", "")  # For cosmetic reasons
+                    raise MyfitnesspalRequestFailed(
+                        f"Unable to submit food to MyFitnessPal: {error}"
+                    )
+            except Exception:
+                logger.warning(f"Unable to submit food to MyFitnessPal: {error}")
```

Comment:
```suggestion
            if not document.xpath(
                # If list is empty there should be no error, could be replaced with assert
                "//*[@id='errorExplanation']/ul/li"
            ):
                return True
                # print("No Error :)")
                # Would like to return FoodItem, but seems that it take
                # to long until the submitted food is available in the DB
                # return self.get_food_search_results("{} {}".format(brand, description))[0]
            else:  # Error occurred
                error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
                error = error.replace("Description ", "")  # For cosmetic reasons
                raise MyfitnesspalRequestFailed(
                    f"Unable to submit food to MyFitnessPal: {error}"
                )
```

I kind of think we should just leave that exception uncaught so the caller can deal with it.

Reply:
ok, added with the next commit.